e2e_playwright/st_markdown_test.py

```diff
@@ -0,0 +1,104 @@
+    """Verifies the correct text content of markdown elements."""
+
+    markdown_elements = app.get_by_test_id("stMarkdown")
+
+    # Assert the text content of each markdown element
+    expect(markdown_elements.nth(0)).to_have_text("This markdown is awesome! 😎")
+    expect(markdown_elements.nth(1)).to_have_text("This <b>HTML tag</b> is escaped!")
+    expect(markdown_elements.nth(2)).to_have_text("This HTML tag is not escaped!")
+    expect(markdown_elements.nth(3)).to_have_text("[text]")
+    expect(markdown_elements.nth(4)).to_have_text("link")
+    expect(markdown_elements.nth(5)).to_have_text("[][]")
+    expect(markdown_elements.nth(6)).to_have_text("Inline math with KaTeX\\KaTeXKATE​X")
+    expect(markdown_elements.nth(7)).to_have_text(
+        "ax2+bx+c=0ax^2 + bx + c = 0ax2+bx+c=0"
+    )
+    expect(markdown_elements.nth(8)).to_have_text("Some header 1")
+    expect(markdown_elements.nth(9)).to_have_text("Some header 2")
+    expect(markdown_elements.nth(10)).to_have_text("Some header 3")
+    expect(markdown_elements.nth(11)).to_have_text("Col1Col2SomeData")
+
+    # Additional checks for specific elements like links
+    expect(markdown_elements.nth(3).locator("a")).to_have_count(0)
+    expect(markdown_elements.nth(4).locator("a")).to_have_attribute("href", "href")
+
+
+def test_markdown_displays_headers_anchors(app: Page):
+    """Displays headers with anchors"""
+
+    h1 = app.locator("h1#some-header-1")
+    h2 = app.locator("h2#some-header-2")
+    h3 = app.locator("h3#some-header-3")
+
+    expect(h1).to_have_count(1)
+    expect(h2).to_have_count(1)
+    expect(h3).to_have_count(1)
+
+    expect(h1).to_have_attribute("id", "some-header-1")
+    expect(h2).to_have_attribute("id", "some-header-2")
+    expect(h3).to_have_attribute("id", "some-header-3")
+
+
+def test_markdown_displays_tables(app: Page, assert_snapshot: ImageCompareFunction):
+    """Displays markdown tables correctly"""
+
+    tables = app.locator(".element-container .stMarkdown table")
+    expect(tables).to_have_count(1)
+    assert_snapshot(tables.first, name="markdown-table-visuals")
+
+
+def test_markdown_displays_long_headers_above_other_elements(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Displays long headers above other elements in the markdown block"""
+
+    long_header = app.locator(
+        "[data-testid='stVerticalBlock'] [data-testid='stVerticalBlock']"
+    ).nth(0)
+    assert_snapshot(long_header, name="long-markdown-header-above-table")
+
+
+def test_markdown_displays_headings_and_markdown_together(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Checks how headings and markdown are displayed when called separately or together and take snapshots."""
+
+    heading_and_markdown_block = app.locator(
+        "[data-testid='stVerticalBlock'] [data-testid='stVerticalBlock']"
+    ).nth(1)
+    assert_snapshot(
+        heading_and_markdown_block, name="heading-and-markdown-combinations"
```

Comment:
nit: The suggested naming pattern for screenshots that are related to a specific command is:
`st_command-test_description`, so in this case: `st_markdown-heading_and_markdown_combinations`. But its more a suggestion than a policy.